### ISSUE TITLE
Build failure: sonarr

### ISSUE THREAD

**Issue by aanderse** [NixOS/nixpkgs#360592]
## Steps To Reproduce

1. attempt to upgrade your server to nixos 24.11
2. build `sonarr` package
3. see error that dotnet package is insecure

## Build log

requires an old version of dotnet (`6.x`) which has security vulnerabilities against it in nixpkgs so it  won't build without permitting those vunerabilities

## Additional context

can we upgrade to dotnet `8` which is LTS?

## Notify maintainers

---

Note for maintainers: Please tag this issue in your PR.

@fadenb @purcell @tie 

and maybe @corngood, if my memory serves me correctly?

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by bobberb** [comment]
https://github.com/NixOS/nixpkgs/blob/nixos-24.11/pkgs/by-name/so/sonarr/package.nix

`
  dotnet-sdk = dotnetCorePackages.sdk_6_0;
  dotnet-runtime = dotnetCorePackages.aspnetcore_6_0;`
  
  Are the offending values

**Comment by aanderse** [comment]
yeah, i tried just replacing them with `8.x` but it seems that build scripts expect `6.x`

looking in upstream i am under the impression this _can_ build with `8.x` but maybe some sort of build time config switch is required

i didn't investigate further than that

**Comment by gepbird** [comment]
Upstream hasn't updated to .NET 8, but some attempt was made: https://github.com/Sonarr/Sonarr/issues/6597 (from here you can check the linked PRs)
Projects from the same maintainers have PRs for .NET 8: https://github.com/Radarr/Radarr/pull/10258 https://github.com/Prowlarr/Prowlarr/pull/2195

It takes a lot of effort for them to properly update to .NET 8, but for us applying some bare minimum patches (perhaps from https://github.com/Sonarr/Sonarr/pull/6983) could work.

**Comment by corngood** [comment]
> why is Sonarr being built rather than using the pre-compiled binaries like Radarr, Prowlarr, etc?

We should always prefer source builds IMO, but someone could add `sonarr-bin` if it's useful.

Also, anyone can still build this package with `NIXPKGS_ALLOW_INSECURE`.

**Comment by aanderse** [comment]
i'll include this snippet here as well, citing [discourse](https://discourse.nixos.org/t/solved-sonarr-is-broken-in-24-11-unstable-aka-how-the-hell-do-i-use-nixpkgs-config-permittedinsecurepackages/56828/3?u=aanderse):

```
  nixpkgs.config.permittedInsecurePackages = [
    "aspnetcore-runtime-6.0.36"
    "aspnetcore-runtime-wrapped-6.0.36"
    "dotnet-sdk-6.0.428"
    "dotnet-sdk-wrapped-6.0.428"
  ];
```

**Comment by purcell** [comment]
Yeah, we used to package the upstream binaries for Sonarr and friends, in fact, but that's not at all the preferred Nixpkgs approach, so we switched to building from source.

**Comment by reinux** [comment]
Something that isn't sonarr is causing the same error for me, and I can't quite make out what it is from the stack trace...

**Comment by kira-bruneau** [comment]
It looks like https://github.com/Sonarr/Sonarr/pull/7443 was merged now, so we can drop the dotnet v6 dependency once sonarr v5 is released!

**Comment by bakerboy448** [comment]
v5 isnt even official beta yet and historically sonarr betas last years....just for key context and to manage expectations

**Comment by gmile** [comment]
I am only NixOS beginner, so asking out of curiosity: should it not be possible to use nix language to override the source of sonarr - to point it the latest sha on GitHub? So that the latest version of Sonarr is pulled and built on the fly, while evaluation the configuration 🤔 (ofc one would need to accept that the un-released Sonarr may be unstable)

**Comment by hughobrien** [comment]
Yes [it's just a bit fiddly](https://nixos.wiki/wiki/Overlays)

**Comment by niklaskorz** [comment]
I think we should update to the tagged version https://github.com/Sonarr/Sonarr/commits/v5.0.0.12 even if it's considered beta. Beta beats insecure dependency.

**Comment by bakerboy448** [comment]
Sonarr v5 is not beta nor released in anyway whatsoever.   As confirmed by both release methods:
- https://github.com/Sonarr/Sonarr/releases
- https://services.sonarr.tv/v1/releases

When v5 is beta presumably an announcement will be made.

**Comment by mrshmllow** [comment]
Anyone who down votes niklaskorz comment, what do you recommend nixpkgs users use? Even enabling insecure packages this package is broken for me.

**Comment by aanderse** [comment]
> Even enabling insecure packages this package is broken for me.

please include details - package and service work fine for me on stable, i would like to know what is broken for you

**Comment by jf-uu** [comment]
I also don't have any issues on unstable (ecd26a469ac56357fd333946a99086e992452b6a). You just need the insecure packages from this comment: https://github.com/NixOS/nixpkgs/issues/360592#issuecomment-2513490613

**Comment by mrshmllow** [comment]
@aanderse 

nixpkgs @ 799ba5bffed04ced7067a91798353d360788b30d

```
...
should_not_be_valid_if_child_of_windows_folder: windows specific test
23:03:39.7 Info: --- Start: NzbDrone.Core.Test.ValidationTests.SystemFolderValidatorFixture.should_not_be_valid_if_set_to_bin_folder --- 
23:03:39.7 Info: --- End: NzbDrone.Core.Test.ValidationTests.SystemFolderValidatorFixture.should_not_be_valid_if_set_to_bin_folder --- 

23:03:39.7 Info: --- Start: NzbDrone.Core.Test.ValidationTests.SystemFolderValidatorFixture.should_not_be_valid_if_set_to_windows_folder --- 
23:03:39.7 Info: --- End: NzbDrone.Core.Test.ValidationTests.SystemFolderValidatorFixture.should_not_be_valid_if_set_to_windows_folder --- 

should_not_be_valid_if_set_to_windows_folder: windows specific test
  Passed should_throw_if_script_is_not_set [123 ms]
  Passed should_throw_if_script_is_null [123 ms]
  Passed should_throw_if_script_path_does_not_exist [119 ms]
  Passed should_not_be_valid_if_child_of_bin_folder [158 ms]
  Skipped should_not_be_valid_if_child_of_windows_folder [40 ms]
  Passed should_not_be_valid_if_set_to_bin_folder [32 ms]
  Skipped should_not_be_valid_if_set_to_windows_folder [26 ms]
  Passed equal_operand [25 ms]
  Passed equal_operand_false [4 ms]
  Passed greater_operand [6 ms]
  Passed Icomparer_greater_test(6,0,0) [10 ms]
  Passed Icomparer_greater_test(5,1,0) [< 1 ms]
  Passed Icomparer_greater_test(5,0,1) [< 1 ms]
  Passed Icomparer_lesser_test(4,5,5) [5 ms]
  Passed Icomparer_lesser_test(5,4,5) [< 1 ms]
  Passed Icomparer_lesser_test(5,5,4) [< 1 ms]
  Passed lesser_operand [7 ms]
  Passed not_equal_operand_false [6 ms]
  Passed not_equal_operand_true [4 ms]
NUnit Adapter 3.17.0.0: Test execution complete

Test Run Failed.
Total tests: 4616
     Passed: 4589
     Failed: 1
    Skipped: 26
 Total time: 31.5983 Minutes
```

I am compiling again with doCheck = false but it does take like an hour to build for aarch64
